pkg/serializer/frame/sanitize.go

```diff
@@ -0,0 +1,67 @@
+package frame
+
+import "bytes"
+
+// Sanitizer is an interface for sanitizing frames. Note that a sanitizer can only do
+// its work correctly if frame actually only contains one frame within.
+type Sanitizer interface {
+	// Sanitize sanitizes the frame in a standardized way for the given
+	// FramingType. If the FramingType isn't known, the Sanitizer can choose between
+	// returning an ErrUnsupportedFramingType error or just returning frame, nil unmodified.
+	// If ErrUnsupportedFramingType is returned, the consumer won't probably be able to handle
+	// other framing types than the default ones, which might not be desired.
+	//
+	// The returned frame should have len == 0 if it's considered empty.
+	Sanitize(ct FramingType, frame []byte) ([]byte, error)
+}
+
+// DefaultSanitizer implements frame sanitation for JSON and YAML.
+//
+// For YAML it removes unnecessary "---" separators, whitespace and newlines.
+// The YAML frame always ends with a newline, unless the sanitized YAML was an empty string, in which
+// case an empty string with len == 0 will be returned.
+//
+// For JSON it sanitizes the JSON frame by removing unnecessary spaces and newlines around it.
+type DefaultSanitizer struct{}
+
+func (DefaultSanitizer) Sanitize(ct FramingType, frame []byte) ([]byte, error) {
+	switch ct {
+	case FramingTypeYAML:
+		return sanitizeYAMLFrame(frame), nil
+	case FramingTypeJSON:
+		return sanitizeJSONFrame(frame), nil
+	default:
+		// Just passthrough
+		return frame, nil
+	}
+}
+
+func sanitizeYAMLFrame(frame []byte) []byte {
+	prevLen := len(frame)
+	for {
+		// Trim spaces
```

Comment:
This comment doesn't add anything to the code?

Reply:
Thanks for the feedback; I'll review my comments and make sure they describe "why" not "what". Indeed, some unnecessary "what" comments have slipped in in a fair amount of places.